signature/cose/envelope.go

```diff
@@ -466,25 +496,47 @@ func parseProtectedHeaders(protected cose.ProtectedHeader, signerInfo *signature
 	signingScheme := signature.SigningScheme(signingSchemeString)
 	signerInfo.SignedAttributes.SigningScheme = signingScheme
 
+	// decode Tag1 Datetime CBOR object into time.Time
+	decMode, err := decOpts.DecMode()
+	if err != nil {
+		return &signature.InvalidSignRequestError{Msg: err.Error()}
+	}
+
 	// populate signerInfo.SignedAttributes.SigningTime
 	signingTimeLabel, ok := signingSchemeTimeLabelMap[signingScheme]
 	if !ok {
 		return &signature.InvalidSignatureError{Msg: "unsupported signingScheme: " + signingSchemeString}
 	}
-	signingTime, ok := protected[signingTimeLabel].(int64)
-	if !ok {
-		return &signature.InvalidSignatureError{Msg: "invalid signingTime under signing scheme: " + signingSchemeString}
+	var signingTime time.Time
+	if isSign {
```

Comment:
The `isSign` is not necessary here since `cbor.RawMessage` type will never be unmarshalled to `interface{}`.

Reply:
Therefore, a `switch` statement will be sufficient. Then we can move all related logic into `parseTime()`.